internal/core/input.rs

```diff
@@ -183,6 +183,17 @@ pub enum KeyEventResult {
     EventIgnored,
 }
 
+/// Represents how an item's focus_event handler dealt with a focus event.
+/// An accepted event results in no further event propagation.
+#[repr(C)]
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum FocusEventResult {
+    /// The event was handled.
+    EventAccepted,
+    /// The event was not handled and should be sent to other items.
+    EventIgnored,
```

Comment:
```suggestion
    /// The item accepted the offer for keyboard focus.
    FocusAccepted,
    /// The item does not accept the keyboard focus. The event should be set to other items.
    FocusRejected,
```

Reply:
I followed the naming conventions use for the other *EventResult.

Reply:
I can see that :-). I'm just wondering if perhaps for this kind of event acceptance/rejection with focus in the name is easier to read. Another option would be to call it `set_focus` instead of `focus_event` and `set_focus` can "fail".

This is a comment orthogonal to the approval. I'm not a big fan of the generic event acceptance/ignore pattern. So if an opportunity to replace generic names with specific names, I jump on it :-)

Reply:
If they're meant to be the same, we could also rename KeyEventResult to just EventResult,  for example.  But i think it make sense to have a different enum with different names, like Simon said.

(`set_focus` would be misleading because this doesn't set the focus, just notify the Item it now has the focus or lost it. I think focus_event is fine)

Reply:
I would be up to change the KeyEventResult later to give that less geenric names;-)